clients/src/test/java/org/apache/kafka/clients/consumer/internals/DefaultEventHandlerTest.java

```diff
@@ -0,0 +1,135 @@
+import java.util.concurrent.LinkedBlockingQueue;
+
+import static org.apache.kafka.clients.consumer.ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG;
+import static org.apache.kafka.clients.consumer.ConsumerConfig.RETRY_BACKOFF_MS_CONFIG;
+import static org.apache.kafka.clients.consumer.ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class DefaultEventHandlerTest {
+    private final Properties properties = new Properties();
+
+    @BeforeEach
+    public void setup() {
+        properties.put(KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+        properties.put(VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+        properties.put(RETRY_BACKOFF_MS_CONFIG, "100");
+    }
+
+    @Test
+    @Timeout(1)
+    public void testBasicPollAndAddWithNoopEvent() {
+        Time time = new MockTime(1);
+        LogContext logContext = new LogContext();
+        SubscriptionState subscriptions = new SubscriptionState(new LogContext(), OffsetResetStrategy.NONE);
+        ConsumerMetadata metadata = newConsumerMetadata(false, subscriptions);
+        MockClient client = new MockClient(time, metadata);
+        ConsumerNetworkClient consumerClient = new ConsumerNetworkClient(logContext, client, metadata, time,
+                100, 1000, 100);
+        BlockingQueue<ApplicationEvent> aq = new LinkedBlockingQueue<>();
+        BlockingQueue<BackgroundEvent> bq = new LinkedBlockingQueue<>();
+        DefaultEventHandler handler = new DefaultEventHandler(
+                time,
+                new ConsumerConfig(properties),
+                aq, bq,
+                subscriptions,
+                metadata,
+                consumerClient);
+        assertTrue(client.active());
+        assertTrue(handler.isEmpty());
+        handler.add(new NoopApplicationEvent("testBasicPollAndAddWithNoopEvent"));
+        while (handler.isEmpty()) {
+            time.sleep(100);
+        }
+        assertTrue(handler.poll().get() instanceof NoopBackgroundEvent);
+        assertFalse(client.hasInFlightRequests()); // noop does not send network request
+        handler.close();
+        assertFalse(client.active());
+    }
+
+    @Test
+    @Timeout(1)
+    public void testRunOnceBackgroundThread() {
+        BlockingQueue<ApplicationEvent> applicationEventQueue = new LinkedBlockingQueue<>();
+        BlockingQueue<BackgroundEvent> backgroundEventQueue = new LinkedBlockingQueue<>();
+        EventHandler eventHandler = new DefaultEventHandler(
+                new RunOnceBackgroundThreadRunnable(applicationEventQueue, backgroundEventQueue),
+                applicationEventQueue,
+                backgroundEventQueue);
+        assertTrue(eventHandler.add(new NoopApplicationEvent("hello-world")));
+        while (eventHandler.isEmpty()) { }
```

Comment:
This is a general comment: I think it worth adding a `poll(timeout)` in the `EventHandler` interface since 1) for testing, we do not need the while loop any more, and 2) in production code, for the caller thread scenarios I think there are cases where a `poll(timeout)` would be useful as well.